analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -121,6 +122,23 @@ class Gradle(
     private val maven = MavenSupport(GradleCacheReader())
 
     override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        // Usually, the Gradle wrapper handles applying system properties from Gradle properties. But as we directly use
+        // the tooling API, we need to manually load Gradle properties and apply any system properties. Limit the lookup
+        // to the current user's Gradle properties file for now.
+        val gradleProperties = getUserHomeDirectory().resolve(".gradle/gradle.properties")
+        if (gradleProperties.isFile) {
+            val systemProperties = gradleProperties.inputStream().use {
+                Properties().apply { load(it) }.filter { (key, _) ->
+                    (key as String).startsWith("systemProp.")
+                }
+            }
+
+            // Apply Gradle properties as system properties, but with the "systemProp." prefix removed.
+            systemProperties.forEach { (key, value) ->
+                System.setProperty((key as String).removePrefix("systemProp."), value as String)
```

Comment:
Would it work to pass the system properties in the `connection.withArguments()` call below with `-D` or `--system-prop` instead? I am a bit concerned, that setting the system properties in the current JVM could cause unwanted side effects.

Reply:
No, that would not work, see https://github.com/gradle/gradle/issues/6825#issuecomment-499470955 (which you upvoted, BTW 😋 ). I could limit this to proxy-related system properties (which is what I had initially in this PR), but then I though that if *any* system properties are defined in `gradle.properties` then *all* of them should apply. Or I could restore the original system properties afterwards.

Reply:
Is it even required to read the proxy settings from the Gradle properties? You could also set them directly when starting ORT, right?

Reply:
No. When analyzing a Gradle project that uses the Gradle wrapper with a version that has not been downloaded before, the wrapper needs proxy settings in order to download the Gradle distribution. This is the only way I got this to work.

Reply:
I'm now setting system properties only temporarily.

Reply:
I think it would be very convenient if we had a proxy setting for ORT itself which would then automatically be applied to all third-party tools.

Reply:
I was also considering this as I was initially thinking the same, but then again projects that build manually just fine should also be successfully analyzed by ORT without setting ORT-specific (proxy) settings, which means this change would be needed anyway.

However, ORT-specific (proxy) settings could allow the other way around: (Foreign) projects that would not build manually could still be analyzed by ORT, which could indeed be convenient.